lib/common/schemas.c

```diff
@@ -90,10 +104,25 @@ pcmk__find_x_0_schema_index(GList *schemas)
     best_node = g_list_nth(schemas, best);
     best_schema = best_node->data;
 
-    /* If this is a singleton list, we're done. */
+    /* If we are unit testing, we don't add the pacemaker-next/none schemas
+     * to the list because we're not using the standard schema adding
+     * functions.  Thus, a singleton list means we're done.
+     *
+     * On the other hand, if we are running as usually, we have those two
+     * schemas added to the list.  A list of length three actually only has
+     * one useful schema.  So we're still done.
+     *
+     * @COMPAT Change this when we stop adding those schemas.
```

Comment:
We'll drop next with 3.0.0, but none will always be around

Reply:
In that case, it would make sense to add none when building the lists in the unit tests so we can ditch the differences.